hexa/plugins/connector_postgresql/models.py

```diff
@@ -0,0 +1,100 @@
+import json
+import uuid
+
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.urls import reverse
+
+from hexa.catalog.models import (
+    CatalogIndex,
+    CatalogIndexPermission,
+    CatalogIndexType,
+)
+
+from hexa.core.models import Permission
+from hexa.core.models.cryptography import EncryptedTextField
+
+
+class PostgresqlDatabaseQuerySet(models.QuerySet):
+    def filter_for_user(self, user):
+        if user.is_active and user.is_superuser:
+            return self
+
+        return self.filter(
+            bucketpermission__team__in=[t.pk for t in user.team_set.all()]
+        )
+
+
+class PostgresqlDatabase(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    hostname = models.CharField(max_length=200)
+    username = models.CharField(max_length=200)
+    password = EncryptedTextField(max_length=200)
```

Comment:
I would prefer that we keep the `Credentials` model pattern - let's discuss it...